subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GradleBuildSmokeTest.groovy

```diff
@@ -41,4 +45,16 @@ class GradleBuildSmokeTest extends AbstractGradleceptionSmokeTest {
         result.task(":distributions-full:binDistributionZip").outcome == TaskOutcome.SUCCESS
         result.task(":distributions-full:binInstallation").outcome == TaskOutcome.SUCCESS
     }
+
+    def "can install Gradle distribution over itself"() {
+        def runner = runner('install', "-Pgradle_installPath=$targetDir", '--stacktrace')
+            .ignoreDeprecationWarnings("https://github.com/gradle/gradle-private/issues/3405")
+
+        when:
+        result = runner.build()
+        result = runner.build()
+
+        then:
+        result.task(":distributions-full:install").outcome == TaskOutcome.SUCCESS
```

Comment:
❔ Should we check here that the distribution has been written to the target directory as well?

Reply:
Great idea, thanks!